src/lexer.jl

```diff
@@ -1038,22 +1021,11 @@ function lex_identifier(l::Lexer{IO_t,T}, c) where {IO_t,T}
     return emit(l, get(kw_hash, h, IDENTIFIER))
 end
 
-# This creates a hash using 5 bit per lower case ASCII char.
-# It checks its input to be between 'a' and 'z' (because only those chars)
-# are valid in keywords, and returns a sentinel value for invalid inputs
-# or when the hash is about to overflow.
-function simple_hash(c, h)
-    h == UInt64(0xff) && return h
-    # only 'a' - 'z' actually need to be hashed
-    'a' <= c <= 'z' || return UInt64(0xff)
-    # catch possible overflow by checking the 10 high bits
-    (h & (UInt64(0x3ff) << (64 - 10))) > 0 && return UInt64(0xff)
-    UInt64(h) << 5 + UInt8(c - 'a' + 1)
-end
+@inline simple_hash(c::Char, h::UInt64) = hash(c, h)
```

Comment:
`simple_hash` currently has the property that it's a perfect hash on the subset of keywords, so identifiers can't collide with keywords.

But I guess using `Base.hash` won't necessarily have this property and https://github.com/JuliaLang/Tokenize.jl/issues/189 could reoccur?

It may be better just to optimize `simple_hash` a bit and see if all the branches can be removed.  Tracking the length of the identifier (in bytes not chars?) as well as this hash would probably make the `0xff` special cases unnecessary.

Reply:
That's a fair point. I pushed a commit that *should* be correct and is only barely slower than `Base.hash`. Would be great if you (or someone else) could also try benchmarking this because timings are somewhat unstable on my machine.